index.js

```diff
@@ -55,6 +54,12 @@ class ClientServiceBase {
      */
     this.ledgerPrivileged = services['ledgerPrivileged'];
 
+    /**
+     * The Signer class provided by Node SDK or Web SDK
+     * @constant
+     */
+    this.signerFactory = services['signerFactory'];
```

Comment:
`this.signer` should  also be created here instead of doing it in each function ?

Reply:
We might not be able to create a signer in the constructor in some cases that the users are not providing private keys, i.e., the registerCertificate and registerFunction cases.
The signer creation is moved to function level. To prevent that we create new signer every time as a function is called, we use the singleton pattern in the function.
```
this.signer = this.signer || this.signerFactory.create(...)
```
The signer is created for the first time and can be used in all subsequent function calls.

Reply:
Right ! got it now. Thanks.